### PR TITLE
Docstrings for storage backend

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

**Comment by Natim** [comment]
:)

File: cliquet/storage/__init__.py [hunk 1]
```diff
@@ -11,28 +11,167 @@ def __init__(self, id_generator=None, *args, **kwargs):
...
+
+        :returns: the record object.
+        :rtype: dict
+        """
         raise NotImplementedError
 
     def update(self, resource, user_id, record_id, record):
+        """Overwrite the record with the specified id.
+
+        If the specified id is not found, the record is created with the
```
Comment (Natim) [inline cliquet/storage/__init__.py]: It should raise a RecordNotFound IMHO

File: cliquet/storage/__init__.py [hunk 2]
```diff
@@ -11,28 +11,167 @@ def __init__(self, id_generator=None, *args, **kwargs):
...
+        """
+
+        :param resource: the record associated resource
+        :type resource: cliquer.resource.BaseResource
+
+        :param user_id: the owner of the record
+        :type user_id: unicode
+
+        :param filters: Optionally filter the records by their attribute.
+            Each filter in this list refers to a field, a value and an
```
Comment (Natim) [inline cliquet/storage/__init__.py]: a tuple of a field, value and comparison.

File: cliquet/storage/__init__.py [hunk 3]
```diff
@@ -11,28 +11,167 @@ def __init__(self, id_generator=None, *args, **kwargs):
...
+
+        :param limit: Optionnally limit the number of records to be retrieved.
+        :type limit: integer
+
+        :param include_deleted: Optionnally include the deleted records that
+            match the filters.
+        :type include_deleted: boolean
+
+        :returns: the limited list of records, and the total number of
+            matching records.
```
Comment (Natim) [inline cliquet/storage/__init__.py]: in the collection (without the deleted ones.)

File: cliquet/storage/__init__.py (hunk 1 above)
Comment (leplatrem) [inline cliquet/storage/__init__.py]: Please open an issue

Comment (Natim) [inline cliquet/storage/__init__.py]: Actually maybe do we need this to work like that to handle the PUT endpoint.

**Pull request merged**